diff --git a/components/retro-go/CMakeLists.txt b/components/retro-go/CMakeLists.txt
index 859baddf..68375061 100644
--- a/components/retro-go/CMakeLists.txt
+++ b/components/retro-go/CMakeLists.txt
@@ -1,5 +1,10 @@
-set(COMPONENT_SRCDIRS ". drivers/audio fonts libs/netplay libs/lodepng")
-set(COMPONENT_ADD_INCLUDEDIRS ". libs/netplay libs/lodepng")
+set(COMPONENT_SRCDIRS ". drivers/audio fonts libs/lodepng")
+set(COMPONENT_ADD_INCLUDEDIRS ". libs/lodepng")
+
+if(RG_ENABLE_NETPLAY)
+    set(COMPONENT_SRCDIRS "${COMPONENT_SRCDIRS} libs/netplay")
+    set(COMPONENT_ADD_INCLUDEDIRS "${COMPONENT_ADD_INCLUDEDIRS} libs/netplay")
+endif()
 if((IDF_VERSION_MAJOR EQUAL 5) AND (IDF_VERSION_MINOR GREATER 2)) # esp-idf 5.3+
     set(COMPONENT_REQUIRES "spi_flash fatfs app_update json nvs_flash esp_adc esp_timer esp_psram esp_wifi esp_http_client esp_driver_ledc driver")
 elseif(IDF_VERSION_MAJOR EQUAL 5) # esp-idf 5.0 - 5.2
diff --git a/retro-core/CMakeLists.txt b/retro-core/CMakeLists.txt
index 1e302d40..7bbd5e25 100644
--- a/retro-core/CMakeLists.txt
+++ b/retro-core/CMakeLists.txt
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 3.5)
 set(COMPONENTS "main retro-go fceumm gnuboy pce-go gw-emulator handy smsplus snes9x")
-set(RG_ENABLE_NETWORKING 0)
+if(NOT DEFINED RG_ENABLE_NETWORKING)
+    set(RG_ENABLE_NETWORKING 0)
+endif()
 include(../base.cmake)
 project(retro-core)
diff --git a/retro-core/components/gnuboy/gnuboy.c b/retro-core/components/gnuboy/gnuboy.c
index 938406b9..63f07471 100644
--- a/retro-core/components/gnuboy/gnuboy.c
+++ b/retro-core/components/gnuboy/gnuboy.c
@@ -6,7 +6,6 @@
 #include <stdlib.h>
 #include <string.h>
 
-
 #define hw GB
 
 // Set in the far future for VBA-M support
@@ -167,7 +166,7 @@ void gnuboy_load_bank(int bank) {
   const size_t OFFSET = bank * BANK_SIZE;
 
   if (!cart.rombanks[bank])
-    cart.rombanks[bank] = malloc(BANK_SIZE);
+    cart.rombanks[bank] = rg_alloc(BANK_SIZE, MEM_SLOW);
 
   if (!cart.romFile)
     return;
